carapace-server/src/main/java/org/carapaceproxy/core/HttpProxyServer.java

```diff
@@ -717,6 +723,10 @@ private void applyDynamicConfiguration(ConfigurationStore newConfigurationStore,
                 proxyRequestsManager.reloadConfiguration(newConfiguration, newBackends.values());
             }
 
+            if(currentConfiguration.isMaintenanceModeEnabled() != newConfiguration.isMaintenanceModeEnabled()) {
```

Comment:
Is this necessary? applyDynamicConfigurationFromAPI should do this

Reply:
I think it is necessary, because otherwise the releoad of the proxyRequestManager is done only when the backend configuration changes.

The only place where maintenance is checked is in the proxyRequestManager. So we have to reload even when the maintenance status changes.